### DISCUSSION TITLE
Stackstorm-Zabbix

### DISCUSSION THREAD

**Discussion by mounika-bokka** [StackStorm/st2#5762] [Packs]
Hi All,

I am trying to do zabbix-stackstorm integration, after i performed this step /opt/stackstorm/packs/zabbix/tools/register_st2_config_to_zabbix.py -z http://zabbix-host/zabbix -u Admin -p zabbix , i came across this error Traceback (most recent call last):
  File "./register_st2_config_to_zabbix.py", line 191, in <module>
    main()
  File "./register_st2_config_to_zabbix.py", line 176, in main
    mediatype_id = register_media_type(client, options, mediatype_id)
  File "./register_st2_config_to_zabbix.py", line 83, in register_media_type
    ret = client.mediatype.create(**params)
  File "/opt/stackstorm/virtualenvs/zabbix/lib/python3.8/site-packages/pyzabbix/api.py", line 80, in fn
    return self.parent.do_request(
  File "/opt/stackstorm/virtualenvs/zabbix/lib/python3.8/site-packages/pyzabbix/api.py", line 250, in do_request
    raise ZabbixAPIException(msg, err['code'])
pyzabbix.api.ZabbixAPIException: ("Error -32602: Invalid params., Media type is missing parameters: name while sending {'jsonrpc': '2.0', 'method': 'mediatype.create', 'params': {'description': 'StackStorm', 'type': '1', 'exec_path': 'st2_dispatch.py', 'exec_params': '-- CHANGE ME : api_url  (e.g. https://st2-node/api/v1)\\n-- CHANGE ME : auth_url (e.g. https://st2-node/auth/v1)\\n-- CHANGE ME : login uername of StackStorm --\\n-- CHANGE ME : login password of StackStorm --\\n{ALERT.SENDTO}\\n{ALERT.SUBJECT}\\n{ALERT.MESSAGE}\\n'}, 'id': '1', 'auth': 'ca5ae3a11f8f9b9027ae71637e1dba7e'}", -32602)

can anyone let me know how to fix this

**Comment by akhilvadagam** [comment]
can anyone let us know if any other step worked for this error

**Comment by nzlosh** [comment]
What version of zabbix are you using?

**Reply by akhilvadagam** [reply]
Zabbix 6.0.10

**Comment by nzlosh** [comment]
According to the pack [documentation](https://github.com/StackStorm-Exchange/stackstorm-zabbix#requirements), it has been tested on version 4.0 and prior:

> Zabbix >3.0. It has been tested with v3.0, v3.2 and v4.0.

My guess is the code isn't compatible with the v6 API. I see you've already commented in the issue: https://github.com/StackStorm-Exchange/stackstorm-zabbix/issues/56 so hopefully someone will able to help you.  If you don't get any response from the original creators of the pack, you'd need patch the code yourself or get someone to patch it for you.
